packages/dds/matrix/src/sparsearray2d.ts

```diff
@@ -83,12 +87,14 @@ export class SparseArray2D<T>
 		return undefined;
 	}
 
-	public get matrixProducer() {
-		// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+	public get matrixProducer(): IMatrixProducer<T> {
+		// Suppression needed to satisfy the interface declaration while also not returning
```

Comment:
Could we return `undefined as unknown as IMatrixProducer<T>`? Would remove the need for the liner overrides, though I think the comment you added would be great to keep.